src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -1335,6 +1336,21 @@ MavlinkMissionManager::parse_mavlink_mission_item(const mavlink_mission_item_t *
 			mission_item->time_inside = mavlink_mission_item->param1;
 			mission_item->loiter_radius = mavlink_mission_item->param3;
 			mission_item->loiter_exit_xtrack = (mavlink_mission_item->param4 > 0);
+
+			// The spec says:
+			// Forward moving aircraft this sets exit xtrack location:
+			// 0 for center of loiter wp, 1 for exit location.
+			// Else, this is desired yaw angle.
+			//
+			// Therefore, we don't touch yaw by setting it NAN for these two cases.
+			if (fabsf(mavlink_mission_item->param4 - 0.0f) < FLT_EPSILON ||
+			    fabsf(mavlink_mission_item->param4 - 1.0f) < FLT_EPSILON) {
```

Comment:
I think this needs to be done by vehicle type. What if a multicopter wants a desired yaw angle of 0 or 1?

Reply:
Ah, just enter 360 :smile: